### ISSUE TITLE
git config --replace-all breaks when repo url has multiple words

### ISSUE THREAD

**Issue by hoprocker** [chef/chef#6248]
When using a git repository with multiple words, ie with the `ext::ssh` protocol, the `git config --replace-all` command breaks. This is only a problem when a) a repository has multiple remotes, or b) the repository name doesn't match the shortname.

This appears in the most recent version of Chef (13.2.20), but [that line](https://github.com/chef/chef/blob/462ce499783771dd54d3cba576e6165643a25b11/lib/chef/provider/git.rb#L201) dates from Apr 29, 2016.

The fix should be very simple -- pass the remote url into the config as a quoted string, as is done with the similar `git_ls_remote()` function.

I also see that the in `Chef::Mixin::ShellOut`, there's a recommendation to generally use shell_out_compact() -- maybe that would work here, as well.

**Comment by coderanger** [comment]
How can a URL have multiple words?

**Comment by hoprocker** [comment]
Good question -- when using git's "external transport" protocol feature, ie when using different credentials for different repositories inline. For example,

```
ext::ssh -i /home/deploy-user/.ssh/all_awesome_repos_key -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no git@github.com %S /GreatOrg/one_awesome_repo
```

Gleaned from [here](https://stackoverflow.com/questions/24024783/checkout-git-repo-with-chef-with-ssh-key/33810160#33810160)

**Comment by coderanger** [comment]
You might be interested in the [`poise-git` cookbook/resource](https://github.com/poise/poise-git) which handles this already.

**Comment by hoprocker** [comment]
Yup, looks like that might cover at least this one scenario; I still feel like this might be worth adding, though, to support this particular edge case (which afaik is a valid usage of git)?

**Comment by tas50** [comment]
Fixed in #7014 which will go out in Chef 14.2
